### DISCUSSION TITLE
keeping consistency

### DISCUSSION THREAD

**Discussion by simonmcnair** [adieyal/sd-dynamic-prompts#690] [General]
if I wanted to have ' man wearing __basiccolour__ shirt and  __basiccolour__ trousers, is there any way to specify that multiple uses of the prompt should be random, but consistent across the prompt ?  In this case, I want them to be blue, rather than the first being blue and the second red, as that could be a colour clash that I don't want ?

**Comment by akx** [comment]
Use an immediately evaluated variable to assign the value once, then reference the variable later in the prompt.

**Reply by simonmcnair** [reply]
Thanks for replying.  How do I do that please ?

**Reply by akx** [reply]
Please see [the documentation](https://github.com/adieyal/sd-dynamic-prompts/blob/main/docs/SYNTAX.md#immediate-evaluation):

E.g.:

> `${season=!__season__} In ${season}, I wear ${season} shirts and ${season} trousers`

will always yield the same `season`.

**Reply by simonmcnair** [reply]
thank you, sorry for wasting your time.

**Reply by akx** [reply]
Not at all! The SYNTAX.md document is a bit hidden, maybe :)
